params/config.go

```diff
@@ -678,7 +678,7 @@ func (c *ChainConfig) IsInterop(time uint64) bool {
 }
 
 func (c *ChainConfig) IsCel2(time uint64) bool {
-	return isTimestampForked(c.Cel2Time, time)
+	return c.IsOptimism() && isTimestampForked(c.Cel2Time, time)
```

Comment:
In which case was this wrong?

When we added the hardfork flag we thought of it as a feature flag for celo specific behaviour. So from my perspective it only enables new transaction types (and more), but is not inherently bound to optimism (that might change though)

Reply:
My understanding was that the transition to cel2 would mean that we are also optimism, because at that point we would be using the op-stack and we would be setting the optimism config.

How would we make the transition to be cel2 with the optimism codebase but not be optimism?